### PR TITLE
Test/316 direct api calls

### PR DESCRIPTION
**Pull request by tilomi**
This is the newly implemented data extraction. The code is not "perfectly" implemented (DRY-Principles may be violated a little bit) however for the sake of faster development of new features i would suggest refactoring and documentation could be done later in the semester. I tested everything it now takes around 6 hours from start to elasticsearch server which is a lot faster then the current data extraction and upload to elastic search. The focus of our team was to improve the performance we did this for the data extraction and the upload to elasticsearch.

### CODE REVIEW

**Comment by BernhardHumm** [comment]
This sounds really good!

Have you tested whether you get the identical results compared to using PyWikiBot?

 

 

 

------------------------------------

 

Von: Tilo Michel [mailto:notifications@github.com] 
Gesendet: Sonntag, 10. Mai 2020 15:21
An: hochschule-darmstadt/openartbrowser <openartbrowser@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Betreff: [hochschule-darmstadt/openartbrowser] Test/316 direct api calls (#342)

 

This is the newly implemented data extraction. The code is not "perfectly" implemented (DRY-Principles may be violated a little bit) however for the sake of faster development of new features i would suggest refactoring and documentation could be done later in the semester. I tested everything it now takes around 6 hours from start to elasticsearch server which is a lot faster then the current data extraction and upload to elastic search. The focus of our team was to improve the performance we did this for the data extraction and the upload to elasticsearch.

  _____  


You can view, comment on, or merge this pull request online at:


  https://github.com/hochschule-darmstadt/openartbrowser/pull/342


Commit Summary


*	Sparql endpoint test
*	Artwork extraction implemented
*	Merge branch 'staging' into test/316-direct-api-calls
*	Crawling for artworks implemented
*	Moved user-config to scripts dir
*	Fixed dimensions, change request logic
*	Refactoring on get_wikidata_items
*	Added script for wikipedia extract extraction
*	Added functions
*	Get country labels and motifs + main subjects
*	Get genres and movements with same function as motifs
*	Different fixes
*	Get artists + movements + refactoring
*	Load artists implemented
*	Location extraction added
*	Get classes implemented
*	Merge remote-tracking branch 'origin/staging' into test/316-direct-api-calls
*	Remove unused sparql queries
*	Renaming + Bug fixes
*	es bulk insert instead of single doc insert
*	Change from art_ontology_crawler.py to new scripts
*	Add data_extraction_statistics.txt to gitignore
*	Add new data extraction to travis.yml


File Changes


*	M .travis.yml <https://github.com/hochschule-darmstadt/openartbrowser/pull/342/files#diff-354f30a63fb0907d4ad57269548329e3>  (2) 
*	M scripts/.gitignore <https://github.com/hochschule-darmstadt/openartbrowser/pull/342/files#diff-1a1dc8484cfee189455006542d0221d8>  (4) 
*	M scripts/Dockerfile <https://github.com/hochschule-darmstadt/openartbrowser/pull/342/files#diff-ffb8e5a2a45c17ab8656e0ba08f9c918>  (8) 
*	A scripts/data_extraction/artwork_ids_query.sparql <https://github.com/hochschule-darmstadt/openartbrowser/pull/342/files#diff-dbe14df43d96e20c0fb53897a994e502>  (8) 
*	A scripts/data_extraction/get_wikidata_items.py <https://github.com/hochschule-darmstadt/openartbrowser/pull/342/files#diff-981137067c32c5b48eca3f0f91354705>  (1122) 
*	A scripts/data_extraction/get_wikipedia_extracts.py <https://github.com/hochschule-darmstadt/openartbrowser/pull/342/files#diff-e1b719cbc7c28ac77f3b7d1afe1452a9>  (300) 
*	M scripts/etl.sh <https://github.com/hochschule-darmstadt/openartbrowser/pull/342/files#diff-875deab2f2e9c970bbf61d5004764a3d>  (5) 
*	M scripts/requirements.txt <https://github.com/hochschule-darmstadt/openartbrowser/pull/342/files#diff-cea6ed0c92488e750d8abb9dfc62abdd>  (1) 
*	M scripts/run_etl.bat <https://github.com/hochschule-darmstadt/openartbrowser/pull/342/files#diff-b49d98f7a7a4387fdcc35ca5313271a4>  (17) 
*	M scripts/upload_to_elasticsearch/elasticsearch_helper.py <https://github.com/hochschule-darmstadt/openartbrowser/pull/342/files#diff-c4af01cb90f1cb1f40443782a3298447>  (41) 


Patch Links:


*	https://github.com/hochschule-darmstadt/openartbrowser/pull/342.patch
*	https://github.com/hochschule-darmstadt/openartbrowser/pull/342.diff

File: scripts/data_extraction/artwork_ids_query.sparql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# QID: QID of an artwork category
+SELECT DISTINCT
+?item
+WHERE {
+    ?cls wdt:P279* $QID .
+    ?item wdt:P31 ?cls;
+    wdt:P18 ?image .
```
Comment (tilomi) [inline scripts/data_extraction/artwork_ids_query.sparql:7]: Image is not mandatory anymore

**Comment by tilomi** [comment]
Yes i tested this. The item counts are nearly the same around 277000 items at the moment (artworks, motifs, etc.). I also tested the frontend with the new data and everything seemed to work correctly. The new crawler also ran yesterday on the staging server. I only encountered one problem which doesn't effect the frontend #343 and i'm not sure if it was detected in the pywikibot implementation before.

File: scripts/data_extraction/artwork_ids_query.sparql (hunk 1 above)
Comment (tilomi) [reply scripts/data_extraction/artwork_ids_query.sparql:7]: #95 The image is required

**Comment by jschurse** [comment]
I had a deeper look at this.
It is a working solution with much better performance than what we have currently. 
The code seems a bit redundant, even bloated around certain spots (e.g. an own method for the extraction of nearly each attribute) which we probably should refactor via the course of this semester.

All in all a pretty solid job! 👍

**Pull request merged**